editoast/src/views/stdcm/mod.rs

```diff
@@ -1,110 +1,108 @@
-    fetch_pathfinding_payload_track_map, parse_pathfinding_payload_waypoints, StepPayload,
-};
-use super::timetable::get_simulated_schedules_from_timetable;
-use super::train_schedule::process_simulation_response;
-use super::train_schedule::projection::Projection;
-use super::train_schedule::simulation_report::{create_simulation_report, SimulationReport};
-
-use crate::schema::utils::geometry::diesel_linestring_to_geojson;
-use chrono::{DateTime, TimeZone, Utc};
-use derivative::Derivative;
-use geos::geojson::Geometry;
-use uuid::Uuid;
-
-use crate::schema::rolling_stock::RollingStockComfortType;
-
-pub fn routes() -> impl HttpServiceFactory {
-    web::scope("/stdcm").service(create)
+crate::routes! {
+    "/stdcm" => {
+        create
+    }
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
-pub struct STDCMRequestPayload {
-    pub infra_id: i64,
-    pub timetable_id: i64,
-    pub start_time: Option<f64>,
-    pub end_time: Option<f64>,
-    pub steps: Vec<StepPayload>,
-    pub rolling_stock_id: i64,
-    pub comfort: RollingStockComfortType,
-    pub maximum_departure_delay: Option<f64>,
-    pub maximum_run_time: f64,
-    pub speed_limit_tags: Option<String>,
+/// An STDCM request
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, ToSchema)]
+struct STDCMRequestPayload {
+    infra_id: i64,
+    timetable_id: i64,
+    start_time: Option<f64>,
+    end_time: Option<f64>,
+    steps: Vec<PathfindingStep>,
+    rolling_stock_id: i64,
+    comfort: RollingStockComfortType,
+    /// By how long we can shift the departure time in seconds
+    #[serde(default = "default_maximum_departure_delay")]
+    #[schema(default = default_maximum_departure_delay)]
+    maximum_departure_delay: f64,
+    /// Specifies how long the total run time can be in seconds
+    #[serde(default = "default_maximum_run_time")]
+    #[schema(default = default_maximum_run_time)]
+    maximum_run_time: f64,
+    /// Train categories for speed limits
+    speed_limit_tags: Option<String>,
+    /// Margin before the train passage in seconds
+    ///
+    /// Enforces that the path used by the train should be free and
+    /// available at least that many seconds before its passage.
     #[serde(default)]
-    pub margin_before: f64,
+    margin_before: f64,
+    /// Margin after the train passage in seconds
+    ///
+    /// Enforces that the path used by the train should be free and
+    /// available at least that many seconds after its passage.
     #[serde(default)]
-    pub margin_after: f64,
-    pub standard_allowance: Option<AllowanceValue>,
+    margin_after: f64,
+    standard_allowance: Option<AllowanceValue>,
 }
 
-#[derive(Serialize, Deserialize, Debug, PartialEq)]
-pub struct STDCMResponse {
-    pub path: STDCMPath,
-    pub simulation: SimulationReport,
+fn default_maximum_departure_delay() -> f64 {
+    2. * 60. * 60. // 2hrs
 }
 
-#[derive(Serialize, Deserialize, Debug, Derivative)]
+fn default_maximum_run_time() -> f64 {
```

Comment:
minor: You might use `const` here which should make the value compute at compile time? Shiny 🎉 
```suggestion
const fn default_maximum_run_time() -> f64 {
```

Reply:
Good catch! I always forget that rust has compile time functions. I changed the functions content because rust doesn't allow floating point arithmetic at compile-time. I've also added the qualifier to other serde(default) functions.

Reply:
Ah, didn't know floating point arithmetic was not supported. But weirdly enough, it is supported in [`const` variables](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=17dc674bf2229994d286bbf425717b9e) 🤔 So maybe instead of adding a comment and removing the calculation (which is kind of useful for readibility), maybe you can do something like this?

```rust
const TWELVE_HOURS_IN_SECONDS: f64 = 12.0_f64 * 60.0_f64 * 60.0_f64;
const fn default_maximum_run_time() -> f64 {
    TWELVE_HOURS_IN_SECONDS
}